### PR TITLE
Split patternfly implementations into 2 different modules

### PR DESCRIPTION
**Pull request by jotak**
That way, dependant streams such as Kiali will not have to pull both pf3 + pf4 when using k-charted
It's also paving the road for potential alternative implementations other than react/pf

Some common code is shared in /web/common

### CODE REVIEW

**Comment by jotak** [comment]
@lucasponce / @israel-hdez / @aljesusg can you please take a look?
I don't think you need to review all the files changes exhaustively, there's no functional change and no code change beside imports.
The goal is simply to split from single module `k-charted-react` to 2 modules `k-charted-pf3` and `k-charted-pf4`, so that Kiali / any consumer of this lib won't have to pull all pf3+4 dependencies because of k-charted.

There's also some common files shared by the two modules, in `web/common`. They're not published as a separate module (to avoid over-engineered architecture) but just added as source & output to tsc compiler, which is working well so far

File: web/pf3/package.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  "files": [
+    "dist"
+  ],
+  "peerDependencies": {
+    "@types/react": ">=16.0.0",
+    "react": ">=16.0.0"
+  },
+  "dependencies": {},
+  "devDependencies": {
+    "@babel/core": "~7.4.3",
```
Comment (mtho11) [inline web/pf3/package.json]: At some point are we going to lock down these dependency versions?

File: web/pf3/tslint.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{
+  "extends": ["tslint-react", "tslint-no-circular-imports"],
+  "rules": {
```
Comment (mtho11) [inline web/pf3/tslint.json]: I wonder if there is any way to keep these in sync with `kiali-ui`?
Maintaining rules in multiple places is difficult.

Comment (mtho11) [reply web/pf3/tslint.json]: I guess we just create a tslint `extends` module for something like `tslint-kiali`.
But AFAIK tslint has just been replaced in kiali-ui with eslint.

Comment (mtho11) [reply web/pf3/tslint.json]: @cfcosta do you want to comment on this?

Comment (mtho11) [reply web/pf3/tslint.json]: I just want to make sure we don't get into a situation where react/typescript in this repo abides by these rules and in the other repo a different set of rules.

**Comment by mtho11** [comment]
So it looks like `tslint` is now deprecated in favor of `eslint`: https://medium.com/palantir/tslint-in-2019-1a144c2317a9

File: web/pf3/package.json (hunk 1 above)
Comment (jotak) [reply web/pf3/package.json]: I don't know. This is the approach we have in Kiali UI, but after having experimenting, I'm questioning this approach for two reasons:

- `yarn.lock` is also ensuring they are locked
- many transitive dependencies have tilde or caret versions, hence even our "locked" dependencies are not so much locked in the end.

**Comment by jotak** [comment]
> I was expecting it to fail the build since kiali-ui rejects to build correctly for sources our of the src directory.

Probably because I'm running tsc directly for build and not react scripts?

**Comment by jotak** [comment]
> So it looks like tslint is now deprecated in favor of eslint: https://medium.com/palantir/tslint-in-2019-1a144c2317a9

Issue created: #26

File: web/pf3/tslint.json (hunk 2 above)
Comment (jotak) [reply web/pf3/tslint.json]: Issue created #29

**Comment by israel-hdez** [comment]
> Probably because I'm running tsc directly for build and not react scripts?

Makes sense!

**Pull request merged**